editor/code_editor.cpp

```diff
@@ -1498,87 +1488,64 @@ void CodeTextEditor::move_lines_down() {
 	text_editor->queue_redraw();
 }
 
-void CodeTextEditor::_delete_line(int p_line, int p_caret) {
-	// this is currently intended to be called within delete_lines()
-	// so `begin_complex_operation` is omitted here
-	text_editor->set_line(p_line, "");
-	if (p_line == 0 && text_editor->get_line_count() > 1) {
-		text_editor->set_caret_line(1, p_caret == 0, true, 0, p_caret);
-		text_editor->set_caret_column(0, p_caret == 0, p_caret);
-	}
-	text_editor->backspace(p_caret);
-	if (p_line < text_editor->get_line_count()) {
-		text_editor->unfold_line(p_line);
-	}
-	text_editor->set_caret_line(p_line, p_caret == 0, true, 0, p_caret);
-}
-
 void CodeTextEditor::delete_lines() {
 	text_editor->begin_complex_operation();
 
-	Vector<int> carets_to_remove;
-
-	Vector<int> caret_edit_order = text_editor->get_caret_index_edit_order();
-	for (int i = 0; i < caret_edit_order.size(); i++) {
-		int c = caret_edit_order[i];
-		int cl = text_editor->get_caret_line(c);
-
-		bool swaped_caret = false;
-		for (int j = i + 1; j < caret_edit_order.size(); j++) {
-			if (text_editor->has_selection(caret_edit_order[j])) {
-				if (text_editor->get_selection_from_line() == cl) {
-					carets_to_remove.push_back(caret_edit_order[j]);
-					continue;
-				}
-
-				if (text_editor->get_selection_to_line() == cl) {
-					if (text_editor->has_selection(c)) {
-						if (text_editor->get_selection_to_line(c) != cl) {
-							text_editor->select(cl + 1, 0, text_editor->get_selection_to_line(c), text_editor->get_selection_to_column(c), c);
-							break;
-						}
-					}
-
-					carets_to_remove.push_back(c);
-					i = j - 1;
-					swaped_caret = true;
-					break;
-				}
-				break;
+	Vector<int> lines = _get_affected_lines();
+	for (int c = 0; c < text_editor->get_caret_count(); c++) {
+		text_editor->deselect(c);
+	}
+	for (const int &line : lines) {
+		// Save caret positions from the line and then remove the carets.
+		Vector<int> caret_columns;
+		Vector<int> carets_to_remove;
+		int indent = text_editor->get_indent_size();
+		for (int c = 0; c < text_editor->get_caret_count(); c++) {
+			if (line == text_editor->get_caret_line(c)) {
+				int visual_column = text_editor->get_caret_column(c);
+				visual_column += MIN(visual_column, text_editor->get_indent_level(line) / indent) * (indent - 1);
+				print_line("From column:", text_editor->get_caret_column(c), "visual column:", visual_column, "indent:", text_editor->get_indent_level(line), "*", text_editor->get_indent_size());
```

Comment:
Oops, some high tech debugging features were left in 😆 . I'll fix this (and another at line 1539) and add that hyphen to re-add at line 1530 as spell check tells me to do, but wait for a little bit if something other comes up.